src/isomorphic/modern/class/ReactBaseClasses.js

```diff
@@ -21,7 +21,28 @@ var warning = require('fbjs/lib/warning');
 /**
  * Base class helpers for the updating state of a component.
  */
+
+const addUpdaterMutationWarn = context => {
+  let updater;
+
+  Object.defineProperty(context, 'updater', {
+    get: () => updater,
+    set(value) {
+      warning(
+        value && value.isValidUpdater,
```

Comment:
Let's maybe prefix `isValidUpdater` with an underscore so people aren't tempted to add it themselves?